repair/row_level.cc

```diff
@@ -591,50 +592,67 @@ static void add_to_repair_meta_for_followers(repair_meta& rm);
 
 future<std::list<repair_row>> to_repair_rows_list(repair_rows_on_wire rows, schema_ptr s, uint64_t seed, repair_master is_master, reader_permit permit, repair_hasher hasher) {
     std::list<repair_row> row_list;
-    lw_shared_ptr<const decorated_key_with_hash> dk_ptr;
-    lw_shared_ptr<mutation_fragment> last_mf;
-    position_in_partition::tri_compare cmp(*s);
-    for (partition_key_and_mutation_fragments& x : rows) {
-        dht::decorated_key dk = dht::decorate_key(*s, x.get_key());
-        if (!(dk_ptr && dk_ptr->dk.equal(*s, dk))) {
-            dk_ptr = make_lw_shared<const decorated_key_with_hash>(*s, dk, seed);
-        }
-        if (is_master) {
-            for (frozen_mutation_fragment& fmf : x.get_mutation_fragments()) {
-                _metrics.rx_row_nr += 1;
-                _metrics.rx_row_bytes += fmf.representation().size();
-                // Keep the mutation_fragment in repair_row as an
-                // optimization to avoid unfreeze again when
-                // mutation_fragment is needed by _repair_writer.do_write()
-                // to apply the repair_row to disk
-                auto mf = make_lw_shared<mutation_fragment>(fmf.unfreeze(*s, permit));
-                auto hash = hasher.do_hash_for_mf(*dk_ptr, *mf);
-                position_in_partition pos(mf->position());
-                row_list.push_back(repair_row(std::move(fmf), std::move(pos), dk_ptr, std::move(hash), is_dirty_on_master::yes, std::move(mf)));
-                co_await coroutine::maybe_yield();
+    std::exception_ptr ex;
+    try {
+        lw_shared_ptr<const decorated_key_with_hash> dk_ptr;
+        lw_shared_ptr<mutation_fragment> last_mf;
+        position_in_partition::tri_compare cmp(*s);
+        for (auto it = rows.begin(); it != rows.end(); it = rows.erase(it)) {
```

Comment:
Let's add a comment saying the motivation for the loop and erase. Normally, we can just loop and not erase for simplicity.

Reply:
sure, no problem